packages/ifpack2/src/Ifpack2_BlockTriDiContainer_def.hpp

```diff
@@ -0,0 +1,2756 @@
+  typedef VectorizationTraits<AVX<double> >::vector_type::value_type type;
+};
+#endif
+
+template <typename Int1, typename Int2, typename View, typename... Indices>
+KOKKOS_INLINE_FUNCTION
+typename std::enable_if<VectorLength<typename View::value_type>::value == 1,
+                        typename View::reference_type>::type
+idx (const View& v, const Kokkos::pair<Int1,Int2>& i0, Indices... indices) {
+  return v(i0.first, indices...);
+}
+
+template <typename Int1, typename Int2, typename View, typename... Indices>
+KOKKOS_INLINE_FUNCTION
+typename std::enable_if<(VectorLength<typename View::value_type>::value > 1),
+                        typename View::value_type::value_type&>::type
+idx (const View& v, const Kokkos::pair<Int1,Int2>& i0, Indices... indices) {
+  return v(i0.first, indices...)[i0.second % VectorLength<typename View::value_type>::value];
+}
+
+// No %, but you have to do the % yourself.
+template <typename Int1, typename Int2, typename View, typename... Indices>
+KOKKOS_FORCEINLINE_FUNCTION
+typename std::enable_if<VectorLength<typename View::value_type>::value == 1,
+                        typename View::reference_type>::type
+fastidx (const View& v, const Kokkos::pair<Int1,Int2>& i0, Indices... indices) {
+  return v(i0.first, indices...);
+}
+
+template <typename Int1, typename Int2, typename View, typename... Indices>
+KOKKOS_FORCEINLINE_FUNCTION
+typename std::enable_if<(VectorLength<typename View::value_type>::value > 1),
+                        typename View::value_type::value_type&>::type
+fastidx (const View& v, const Kokkos::pair<Int1,Int2>& i0, Indices... indices) {
+  return v(i0.first, indices...)[i0.second];
+}
+
+} // namespace Batched
+} // namespace Details
+
+namespace BlockTriDiContainerDetails {
+
+// In what follows, AScalar may be std::complex, while Scalar may be
+// Kokkos::complex.
+
+template <typename Scalar>
+inline void copy (const int n, const Scalar* b, const int ldb, Scalar* y, const int ldy,
+                  const int nvec) {
+  int vi = 0;
+  for (;;) {
+#ifdef KOKKOS_HAVE_PRAGMA_IVDEP
+#   pragma ivdep
+#endif
+#ifdef KOKKOS_ENABLE_PRAGMA_UNROLL
+#   pragma unroll
+#endif
+    for (int i = 0; i < n; ++i)
+      y[i] = b[i];
+    ++vi;
+    if (vi == nvec) break;
+    b += ldb;
+    y += ldy;
+  }
+}
+
+template <int BSZ, typename Scalar>
+inline void copy (const int bsz, const Scalar* const x, const int ldx,
+                  Scalar* const y, const int ldy, const int nvec) {
+  if (BSZ == 0)
+    copy(bsz, x, ldx, y, ldy, nvec);
+  else
+    copy(BSZ, x, ldx, y, ldy, nvec);
+}
+
+template <int sign, typename Scalar, typename AScalar>
+inline void gemv (const AScalar* const A, const int n, const Scalar* x, const int ldx, Scalar* y,
+                  const int ldy, const int nvec) {
+  int vi = 0;
+  for (;;) {
+    for (int r = 0; r < n; ++r) {
+      Scalar accum = 0;
+      const AScalar* const Ar = A + r*n;
+#ifdef KOKKOS_HAVE_PRAGMA_IVDEP
```

Comment:
Is this for our use outside Kokkos ? I don't see it in their config files. Same comment goes for PRAGMA_UNROLL

Reply:
@srajama1, good catch, thanks. Looks like the pattern has changed to KOKKOS_ENABLE_PRAGMA_* in Kokkos_Macros.hpp. We need to check in KokkosKernels whether we're doing it right. I'll change the ifdefs here.